tests/e2e/specs/shopper/filter-products-by-stock.test.ts

```diff
@@ -0,0 +1,202 @@
+import {
+	createNewPost,
+	deleteAllTemplates,
+	insertBlock,
+	switchUserToAdmin,
+	publishPost,
+} from '@wordpress/e2e-test-utils';
+import { selectBlockByName } from '@woocommerce/blocks-test-utils';
+import {
+	BASE_URL,
+	goToTemplateEditor,
+	openBlockEditorSettings,
+	saveTemplate,
+	useTheme,
+	waitForCanvas,
+} from '../../utils';
+
+const block = {
+	name: 'Filter Products by Stock',
+	slug: 'woocommerce/stock-filter',
+	class: '.wc-block-stock-filter',
+	selectors: {
+		editor: {
+			filterButtonToggle: "//label[text()='Filter button']",
+		},
+		frontend: {
+			productsList: '.wc-block-grid__products > li',
+			classicProductsList: '.products.columns-3 > li',
+			filter: 'label[for=outofstock]',
+			submitButton: '.wc-block-components-filter-submit-button',
+		},
+	},
+	urlSearchParamWhenFilterIsApplied: '?filter_stock_status=outofstock',
+	foundProduct: 'Woo Single #3',
+};
+
+const waitForAllProductsBlockLoaded = () =>
+	page.waitForSelector( selectors.frontend.productsList + '.is-loading', {
+		hidden: true,
+	} );
+
+const goToShopPage = () =>
+	page.goto( BASE_URL + '/shop', {
+		waitUntil: 'networkidle0',
+	} );
+
+const { selectors } = block;
```

Comment:
Can we move it up? It's the same issue that I mentioned in the final comment. We rely on `selectors` being hoisted in the `waitForAllProductsBlockLoaded` util method.